Bio/Align/Applications/_Guidance2.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 by Rob Gilmore and Shaurita Hutchins. All rights reserved.
+# Based on ClustalOmega wrapper copyright 2011 by Andreas Wilm.
+#
+# Wrapper for Guidance2 by Rob Gilmore (2017).  http://guidance.tau.ac.il/ver2/
+# Used _ClustalOmega.py as template.
+#
+# This code is part of the Biopython distribution and governed by its
+# license.  Please see the LICENSE file that should have been included
+# as part of this package.
+"""Command line wrapper for the multiple alignment program, GUIDANCE2.
+It weights, filters or masks unreliably aligned positions in multiple sequence alignments.
+"""
+
+from __future__ import print_function
+
+import os
+from Bio.Application import _Option, AbstractCommandline
+
+
+class Guidance2Commandline(AbstractCommandline):
+    u"""Command line wrapper for GUIDANCE2.
+
+    http://guidance.tau.ac.il/ver2/
+
+    Example:
+    --------
+
+    >>> from Bio.Align.Applications import Guidance2Commandline
+    >>> import os
+    >>> seqFile = "HTR1A.ffn"
+    >>> msaProgram = "CLUSTALW"
+    >>> seqType = "codon"
+    >>> outDir = Path(os.getcwd)
+    >>> Guidance2_cline = Guidance2Commandline(seqFile, msaProgram, seqType, str(outDir), bootstraps=20, seqCutoff=0.63,\
+                                    colCutoff=0.9, outOrder='as_input', dataset='HTR1A')
+    >>>> print(Guidance2_cline)
+    perl guidance.pl --seqFile HTR1A.ffn --msaProgram CLUSTALW --seqType codon --outDir C://Users/rgilmore/HTR1A
+    --bootstraps 20 --seqCutoff 0.63 --colCutoff 0.9 --outOrder as_input --dataset HTR1A
+
+
+    You would typically run the command line with Guidance2_cline() or via
+    the Python subprocess module, as described in the Biopython tutorial.
+
+    Citation:
+    ---------
+    Sela, I., Ashkenazy, H., Katoh, K. and Pupko, T. (2015)
+    GUIDANCE2: accurate detection of unreliable alignment regions accounting for the uncertainty of multiple parameters.
+    Nucleic Acids Research, 2015 Jul 1; 43 (Web Server issue): W7-W14.; doi: 10.1093/nar/gkq443
+
+    Landan, G., and D. Graur. (2008).
+    Local reliability measures from sets of co-optimal multiple sequence alignments.
+    Pac Symp Biocomput 13:15-24
+    """
+
+    def __init__(self, cmd="perl guidance.pl", **kwargs):
```

Comment:
This line bothers me, ``cmd="perl guidance.pl"`` - the assumption would be this is a single path to a binary - we may not currently handle spaces, but I would expect them to be escaped/quoted as part of the binary's filename.

Can your tool not provide an executable script? You might call it ``guidance`` or simply make ``guidance.pl`` executable with a perl hashbang line? However, I might call it something less likely to clash with other scrips, maybe ``guidance_alignment_confidence``?

(I'm not a perl expert but there must be standard mechanisms for installing Perl scripts intended to be executed directly)